src/main/webapp/app/markdown-editor/markdown-editor.component.ts

```diff
@@ -227,27 +254,17 @@ export class MarkdownEditorComponent implements AfterViewInit {
     }
 
     /**
-     * @function parse
-     * @desc Check if domainCommands are contained within the text to decide how to parse the text
-     *       1. If no domainCommands are contained parse markdown to HTML and emit the result to the parent component
-     *       2. Otherwise create an array containing all domainCommands identifier passed on from the client,
-     *       3. Create a copy of the markdown text
-     *       4. Create the regEx Expression which searches for the domainCommand identifier
-     *       5. Go through the copy of the markdown text until it is empty and split it as soon as a domainCommand identifier is found into [command]
-     *           5a. One command can contain text over several lines
-     *           5b. All the text between two identifiers is mapped to the first identifier
-     *       6. Reduce the copy by the length of the command
-     *       7. Call the parseLineForDomainCommand for command and save it into content
-     *       8. Emit the content to parent component to assign the values of the array to the right attributes
+     * Parses markdown to generate a preview if the standard preview is used and/or searches for domain command identifiers.
+     * Will emit events for both the generated preview and domain commands.
+     *
      */
     parse(): void {
-        /** check if domainCommands are passed on from the parent component */
-        if (this.domainCommands == null || this.domainCommands.length === 0) {
```

Comment:
I think we have a general issue here with what we call domain commands.

E.g. when programming exercises use a special command to parse tasks, it would be a domain command.
However this would mean that we can't use the default html preview of the markdown editor.
It would be better to provide reusable showdown extensions for domain specific commands.